swift-sdk/IterableAPI.swift

```diff
@@ -86,6 +88,66 @@ import UIKit
         }
     }
     
+    /**
+     * Handles a Universal Link
+     * For Iterable links, it will track the click and retrieve the original URL,
+     * pass it to `IterableURLDelegate` for handling
+     * If it's not an Iterable link, it just passes the same URL to `IterableURLDelegate`
+     *
+     - parameter url: the URL obtained from `UserActivity.webpageURL`
+     - returns: true if it is an Iterable link, or the value returned from `IterableURLDelegate` otherwise
+     */
+    @objc(handleUniversalLink:)
+    @discardableResult
+    public static func handle(universalLink url: URL) -> Bool {
+        return internalImplementation?.handleUniversalLink(url) ?? false
+    }
+    
+    /// This will send the device attribute to the back end when registering the device.
+    ///
+    /// - Parameters:
+    /// - name: The device attribute name
+    /// - value:    The device attribute value
+    @objc(setDeviceAttribute:value:)
+    public static func setDeviceAttribute(name: String, value: String) {
+        internalImplementation?.setDeviceAttribute(name: name, value: value)
+    }
+    
+    /// Remove a device attribute set earlier.
+    ///
+    /// - Parameters:
+    /// - name: The device attribute name
+    @objc(removeDeviceAttribute:)
+    public static func removeDeviceAttribute(name: String) {
+        internalImplementation?.removeDeviceAttribute(name: name)
+    }
+    
+    /// Use this property for getting and showing in-app messages.
+    /// This property has no meaning if IterableAPI has not been initialized using
+    /// IterableAPI.initialize
+    /// ```
+    /// - IterableAPI.inAppManager.getMessages()
+    /// - IterableAPI.inAppManager.show(message: message, consume: true)
+    /// ```
+    public static var inAppManager: IterableInAppManagerProtocol {
+        guard let internalImplementation = internalImplementation else {
+            ITBError("IterableAPI is not initialized yet. In-apps will not work now.")
+            assertionFailure("IterableAPI is not initialized yet. In-apps will not work now.")
+            return EmptyInAppManager()
+        }
+        
+        return internalImplementation.inAppManager
+    }
+    
+    // MARK: Private and Internal
+    
+    static var internalImplementation: IterableAPIInternal?
+    
+    private override init() { super.init() }
+}
+
+// API REQUEST CALLS
+extension IterableAPI {
```

Comment:
I am not sure I like this. We have pretty much moved the entire implementation to an extension. That would be good if in the future we want to extract an interface/protocol.

I do like the separation into `deprecated` extension though.

Reply:
I agree it looks a little weird, but I think the separation is still very helpful, if only because it's easier to mentally model what's an API request (and of course there are a lot of those) and what's SDK specific functionality; previously, the way it was laid out made it difficult to make that distinction. Who knows what our exact future plans are for SDK functionality; that section will almost certainly grow on its own with new features. So that's why I'd like to be very clear in the separation despite the odd sizes.

Reply:
Frankly, I have not seen this type of design elsewhere. It is kind of a code smell to me. Usually extensions are there for protocol conformance or to add functionality to a class. But in this case pretty much the entire class is the extension.

Reply:
See comments about class extensions here
https://docs.swift.org/swift-book/LanguageGuide/Extensions.html

Reply:
Hmm, how else could this be structured? I couldn't think of another way to better separate out sections within the same class. I agree that it's not protocol conformance or new functionality, but I still think this separation is very useful.